src/peer_select.cc

```diff
@@ -572,58 +743,139 @@ PeerSelector::selectPinned()
 void
 PeerSelector::selectSomeNeighbor()
 {
-    CachePeer *p;
-    hier_code code = HIER_NONE;
-    assert(entry->ping_status == PING_NONE);
-
-    if (direct == DIRECT_YES) {
-        entry->ping_status = PING_DONE;
-        return;
-    }
+    assert(entry);
+    assert(direct != DIRECT_YES);
 
 #if USE_CACHE_DIGESTS
-    if ((p = neighborsDigestSelect(request))) {
-        if (neighborType(p, request->url) == PEER_PARENT)
-            code = CD_PARENT_HIT;
-        else
-            code = CD_SIBLING_HIT;
-    } else
+    neighborsDigestSelect(this, request.getRaw());
 #endif
-        if ((p = netdbClosestParent(request))) {
-            code = CLOSEST_PARENT;
-        } else if (peerSelectIcpPing(request, direct, entry)) {
-            debugs(44, 3, "Doing ICP pings");
-            ping.start = current_time;
-            ping.n_sent = neighborsUdpPing(request,
-                                           entry,
-                                           HandlePingReply,
-                                           this,
-                                           &ping.n_replies_expected,
-                                           &ping.timeout);
-
-            if (ping.n_sent == 0)
-                debugs(44, DBG_CRITICAL, "WARNING: neighborsUdpPing returned 0");
-            debugs(44, 3, ping.n_replies_expected <<
-                   " ICP replies expected, RTT " << ping.timeout <<
-                   " msec");
-
-            if (ping.n_replies_expected > 0) {
-                entry->ping_status = PING_WAITING;
-                eventAdd("PeerSelector::HandlePingTimeout",
-                         HandlePingTimeout,
-                         this,
-                         0.001 * ping.timeout,
-                         0);
-                return;
-            }
+    netdbClosestParent(this, request.getRaw());
+
+    planNextStep(DoStartPing, "consider pinging");
+}
+
+static void
+checkNeighborToPingAccessWrapper(allow_t answer, void *data)
```

Comment:
Please use cons for `answer`. I see other, similar cases to fix.

Reply:
It is an ACLCB callback. it is defined as:
  typedef void ACLCB(allow_t, void *);